benches/benches.rs

```diff
@@ -36,25 +39,95 @@ fn criterion_benchmark(c: &mut Criterion) {
+
+    let factorial_function = factorial_runner
+        .find_function("main")
+        .expect("failed to find main factorial function");
+    let fibonacci_function = fibonacci_runner
+        .find_function("main")
+        .expect("failed to find main fibonacci function");
+    let logistic_map_function = logistic_map_runner
+        .find_function("main")
+        .expect("failed to find main logistic map function");
+
+    {
+        let mut factorial_group = c.benchmark_group("factorial_2M");
+
+        factorial_group.bench_function("Cached JIT", |b| {
+            b.iter(|| jit_factorial.invoke_dynamic(factorial_function_id, &[], Some(u128::MAX)));
+        });
+        factorial_group.bench_function("Cached AOT", |b| {
+            b.iter(|| aot_factorial.invoke_dynamic(factorial_function_id, &[], Some(u128::MAX)));
+        });
+
+        factorial_group.bench_function("VM", |b| {
+            b.iter(|| {
+                factorial_runner.run_function_with_starknet_context(
+                    factorial_function,
+                    &[],
+                    Some(usize::MAX),
+                    StarknetState::default(),
+                )
+            });
+        });
+
+        factorial_group.finish();
+    }
+
+    {
+        let mut fibonacci_group = c.benchmark_group("fibonacci_2M");
+
+        fibonacci_group.bench_function("Cached JIT", |b| {
+            b.iter(|| jit_fibonacci.invoke_dynamic(fibonacci_function_id, &[], Some(u128::MAX)));
+        });
+        fibonacci_group.bench_function("Cached AOT", |b| {
+            b.iter(|| aot_fibonacci.invoke_dynamic(fibonacci_function_id, &[], Some(u128::MAX)));
+        });
+        fibonacci_group.bench_function("VM", |b| {
+            b.iter(|| {
+                fibonacci_runner.run_function_with_starknet_context(
+                    fibonacci_function,
+                    &[],
+                    Some(usize::MAX),
+                    StarknetState::default(),
+                )
+            });
+        });
+
+        fibonacci_group.finish();
+    }
+
+    {
+        let mut logistic_map_group = c.benchmark_group("logistic_map");
+
+        logistic_map_group.bench_function("Cached JIT", |b| {
+            b.iter(|| {
+                jit_logistic_map.invoke_dynamic(logistic_map_function_id, &[], Some(u128::MAX))
+            });
+        });
+
+        logistic_map_group.bench_function("Cached AOT", |b| {
+            b.iter(|| {
+                aot_logistic_map.invoke_dynamic(logistic_map_function_id, &[], Some(u128::MAX))
+            });
+        });
+
+        logistic_map_group.bench_function("VM", |b| {
+            b.iter(|| {
+                logistic_map_runner.run_function_with_starknet_context(
+                    logistic_map_function,
+                    &[],
+                    Some(usize::MAX),
+                    StarknetState::default(),
+                )
```

Comment:
The VM is stateful, so you should probably clone the runner before execution. There are a few methods in criterion that allow that kind of setup.

Reply:
This may or may not apply entirely given we're using the `cairo-lang` version rather than ours directly.

Reply:
I reviewed `cairo-lang`'s runner and I don't think this applies, as a VM is created on each execution:

SierraCasmRunner: https://github.com/starkware-libs/cairo/blob/2203a47f8a098cd4718d03bd109ca014049419e7/crates/cairo-lang-runner/src/lib.rs#L162-L178
Run Method: https://github.com/starkware-libs/cairo/blob/2203a47f8a098cd4718d03bd109ca014049419e7/crates/cairo-lang-runner/src/lib.rs#L459-L460